js.go

```diff
@@ -1838,7 +1838,7 @@ func (sub *Subscription) scheduleFlowControlResponse(reply string) {
 func (sub *Subscription) activityCheck() {
 	sub.mu.Lock()
 	jsi := sub.jsi
-	if jsi == nil {
+	if jsi == nil || sub.closed {
```

Comment:
Sorry about that.. So now, since here we exit if the subscription is closed, the capture down there where we do:
```
closed := sub.closed
sub.mu.Unlock()

if !active && !closed {
..
```
is silly... we can simply by removing `closed := sub.closed` and simply have `if !active {..`